src/wrapper/index.ts

```diff
@@ -113,6 +114,157 @@ class Log10Wrapper {
       return response;
     };
   }
+
+  async *wrappedBedrockResponse(response: any, request: any) {
+    try {
+      let buffer = "";
+      for await (const chunk of response.body) {
+        const decoded = new TextDecoder().decode(chunk.chunk?.bytes);
+        const parsed = JSON.parse(decoded);
+        if (parsed.type === 'content_block_delta') {
+          buffer += parsed.delta?.text || '';
+        }
+        yield chunk;
+      }
+
+      // Transform the request/response for logging
+      const requestBody = JSON.parse(
+        typeof request.input.body === 'string' 
+          ? request.input.body 
+          : new TextDecoder().decode(
+              request.input.body instanceof Uint8Array 
+                ? request.input.body 
+                : new Uint8Array(request.input.body as ArrayBuffer)
+            )
+      );
+      
+      const transformedRequest = {
+        model: request.input.modelId,
+        messages: requestBody.messages || [{
+          role: "user",
+          content: requestBody.prompt
+        }],
+        max_tokens: requestBody.max_tokens,
+        temperature: requestBody.temperature,
+      };
+
+      if (requestBody.system) {
+        transformedRequest.messages.unshift({
+          role: "system",
+          content: requestBody.system
+        });
+      }
+
+      const transformedResponse = {
+        id: `stream-${Date.now()}`,
+        object: "chat.completion",
+        created: Date.now(),
+        model: request.input.modelId,
+        choices: [{
+          message: {
+            role: "assistant",
+            content: buffer
+          },
+          logprobs: null,
+          index: 0,
+          finish_reason: "stop"
+        }],
+        usage: {
+          prompt_tokens: -1,
+          completion_tokens: -1,
+          total_tokens: -1
+        }
+      };
+
+      this.logCompletion({
+        request: transformedRequest,
+        response: transformedResponse,
+      });
+    } catch (error) {
+      console.error("Error in wrappedBedrockResponse:", error);
+    }
+  }
+
+  wrapBedrock(client: any) {
+    const originalSend = client.send;
+    
+    client.send = async (command: any, ...args: any[]) => {
+      if (!(command instanceof InvokeModelCommand) && 
+          !(command instanceof InvokeModelWithResponseStreamCommand)) {
```

Comment:
Both of the examples don't seem to satisfy these restraints such that a completion would be logged from one of the paths below this code.
Even though they use `InvokeModelCommand` and `InvokeModelWithResponseStreamCommand`

Reply:
This may be due to the way I am testing locally - seems javascript's prototypal inheritance doesn't play nice with multiple instances of the same module
i.e. my test package that references the local package has the same `@aws-sdk/client-bedrock-runtime` version but the  `InvokeModelCommand` instances won't satisfy the `instanceof` call

Reply:
Commenting that check out, it does log the completion 
![image](https://github.com/user-attachments/assets/aa752c31-60cb-4627-951b-6bb8def2bff4)

Reply:
Streaming example:

![image](https://github.com/user-attachments/assets/bc24e57a-10ca-4958-be1b-b9b96eb1f4ff)